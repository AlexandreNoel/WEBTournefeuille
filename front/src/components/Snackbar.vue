<template>
  <v-snackbar
    v-model="snackbar"
    :color="color"
    :timeout="timeout"
    bottom
    left
  >
    {{ text }}
  </v-snackbar>
</template>

<script>
import Toaster from '@/toaster.js'

export default {
  name: 'Snackbar',

  data () {
    return {
      snackbar: false,
      color: null,
      text: null,
      timeout: 6000
    }
  },

  created () {
    Toaster.$on('toast', this.toast)
    Toaster.$on('info', this.info)
    Toaster.$on('success', this.success)
    Toaster.$on('warning', this.warning)
    Toaster.$on('error', this.error)
  },

  methods: {
    async toast (snack = {}) {
      this.snackbar = false
      await this.$nextTick()

      this.snackbar = true
      this.color = snack.color
      this.text = snack.text
    },

    info (msg) {
      this.toast({ color: 'info', text: msg })
    },

    success (msg) {
      this.toast({ color: 'success', text: msg })
    },

    warning (msg) {
      this.toast({ color: 'warning', text: msg })
    },

    error (msg) {
      this.toast({ color: 'error', text: msg })
    }
  }
}
</script>
