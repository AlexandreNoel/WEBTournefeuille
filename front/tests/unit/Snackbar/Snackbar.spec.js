import { shallowMount } from '@vue/test-utils'
import Snackbar from '@/components/Snackbar'

describe('Snackbar', () => {
  it('should render properly', () => {
    const wrapper = shallowMount(Snackbar)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render properly when info is call', async () => {
    const wrapper = shallowMount(Snackbar)

    wrapper.vm.info('Info...')
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render properly when success is call', async () => {
    const wrapper = shallowMount(Snackbar)

    wrapper.vm.success('Success...')
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render properly when warning is call', async () => {
    const wrapper = shallowMount(Snackbar)

    wrapper.vm.warning('Warning...')
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render properly when error is call', async () => {
    const wrapper = shallowMount(Snackbar)

    wrapper.vm.error('Error...')
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })
})
