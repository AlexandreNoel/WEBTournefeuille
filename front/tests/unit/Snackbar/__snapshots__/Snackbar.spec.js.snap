// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar should render properly 1`] = `
<vsnackbar-stub timeout="6000" bottom="" left="">

</vsnackbar-stub>
`;

exports[`Snackbar should render properly when error is call 1`] = `
<vsnackbar-stub timeout="6000" bottom="" left="" color="error">
  Error...
</vsnackbar-stub>
`;

exports[`Snackbar should render properly when info is call 1`] = `
<vsnackbar-stub timeout="6000" bottom="" left="" color="info">
  Info...
</vsnackbar-stub>
`;

exports[`Snackbar should render properly when success is call 1`] = `
<vsnackbar-stub timeout="6000" bottom="" left="" color="success">
  Success...
</vsnackbar-stub>
`;

exports[`Snackbar should render properly when warning is call 1`] = `
<vsnackbar-stub timeout="6000" bottom="" left="" color="warning">
  Warning...
</vsnackbar-stub>
`;
